airflow/operators/python.py

```diff
@@ -501,27 +555,152 @@ def _iter_serializable_context_keys(self):
+            **kwargs,
+        )
+
+    def execute_callable(self):
+        python_path = Path(self.python)
+        if not python_path.exists():
+            raise ValueError(f"Python Path '{python_path}' must exists")
+        if not python_path.is_file():
+            raise ValueError(f"Python Path '{python_path}' must be a file")
+        if not python_path.is_absolute():
+            raise ValueError(f"Python Path '{python_path}' must be an absolute path.")
+        python_version_as_list_of_strings = self._get_python_version_from_venv()
+        if (
+            python_version_as_list_of_strings
+            and str(python_version_as_list_of_strings[0]) != str(sys.version_info.major)
+            and (self.op_args or self.op_kwargs)
+        ):
+            raise AirflowException(
+                "Passing op_args or op_kwargs is not supported across different Python "
+                "major versions for ExternalPythonOperator. Please use string_args."
+                f"Sys version: {sys.version_info}. Venv version: {python_version_as_list_of_strings}"
+            )
+        with TemporaryDirectory(prefix='tmd') as tmp_dir:
+            tmp_path = Path(tmp_dir)
+            return self._execute_python_callable_in_subprocess(python_path, tmp_path)
+
+    def _get_virtualenv_path(self) -> Path:
+        return Path(self.python).parents[1]
+
+    def _get_python_version_from_venv(self) -> List[str]:
+        try:
+            result = subprocess.check_output([self.python, "--version"], text=True)
+            return result.strip().split(" ")[-1].split(".")
+        except Exception as e:
+            raise ValueError(f"Error while executing {self.python}: {e}")
+
+    def _get_airflow_version_from_venv(self) -> Optional[str]:
+        try:
+            result = subprocess.check_output(
+                [self.python, "-c", "from airflow import version; print(version.version)"], text=True
+            )
+            venv_airflow_version = result.strip()
+            if venv_airflow_version != airflow_version:
+                raise AirflowConfigException(
+                    f"The version of Airflow installed in the virtualenv {self._get_virtualenv_path()}: "
+                    f"{venv_airflow_version} is different than the runtime Airflow version: "
+                    f"{airflow_version}. Make sure your environment has the same Airflow version "
+                    f"installed as the Airflow runtime."
                 )
-                raise
+            return venv_airflow_version
+        except Exception as e:
+            self.log.info("When checking for Airflow installed in venv got %s", e)
+            self.log.info(
+                f"This means that Airflow is not properly installed in the virtualenv "
+                f"{self._get_virtualenv_path()}. Airflow context keys will not be available. "
+                f"Please Install Airflow {airflow_version} in your venv to access them."
+            )
```

Comment:
One thought here: Part of the primary reason for using an external Python is to be able to run a task which has conflicts with Core Airflow (for instance some dbt-core libs are tricky to import with airflow right now I think) so is it worth being able to silence this warning somehow?

Reply:
I've added "expect_airflow" and "expect_pendulum" and raise those logs to warning levels if someone sets it to True, but skip the logs entirely otherwise.

Reply:
The "expect_airflow" tunre out  also to be  useful for PyhonVirtualenv, because then it will skip attempting to import airflow plugins if not set (i slightly modified the script template). That seems to be more consistent (and I left default value to be true in both, but users can disable it)